### ISSUE TITLE
[Fastrack] propose a few obvious changes to the composer and chatroom view

### ISSUE THREAD

**Issue by ear-dev** [WideChat/Rocket.Chat#361]

**Comment by ear-dev** [comment]
- sketches or wireframes

- create a table comparing the native mobile client with our PWA in order to propose changes and verify feasibility.

**Comment by KashishJagyasi** [comment]
Analysing all the collected data including TCTSY survey, most of the people were having difficulties in locating threads and discussions. So I've mainly tried to solve that problem.
![Frame 52 (2)](https://user-images.githubusercontent.com/47213420/88542493-240e1180-d034-11ea-8720-f6415b6e9fd8.png)

**Comment by ear-dev** [comment]
@KashishJagyasi This is nice.....  thanks for putting all this together.

@bizzbyster can you take a look please?  We can turn these into stories for the team and get started on it asap if you like.....

**Comment by ear-dev** [comment]
@KashishJagyasi can you link to this please in your wiki #2?  Thanks.

Also.....  is the livechat widget work represented properly on that wiki right now.......  with the changes you made after the back and forth with Ona?

**Comment by KashishJagyasi** [comment]
Yeah, sorry I updated the wiki now. About the livechat widget, I think Ona missed my mail or that update. I think I'll mail her again. Thanks!
@ear-dev

**Comment by bizzbyster** [comment]
Why do discussions not just take place in rooms dedicated to related topics?

**Comment by bizzbyster** [comment]
![image](https://user-images.githubusercontent.com/4001792/88589085-9bd53f80-d026-11ea-8a54-247544e8e76e.png)

**Comment by bizzbyster** [comment]
why not just put the camera and the discussion button behind the +?

Also, what does it look like as I'm typing?.

**Comment by KashishJagyasi** [comment]
> why not just put the camera and the discussion button behind the +?

These are just versions or possibilities to show how it will look like. I've put these behind the + in the first screen 
@bizzbyster 
![Frame 54](https://user-images.githubusercontent.com/47213420/88664226-63069a80-d0fa-11ea-8539-62f0569616d5.png)

**Comment by KashishJagyasi** [comment]
> Why do discussions not just take place in rooms dedicated to related topics?

We try to use these issues and the wiki space so that all the work remains properly documented with each topic separately. But if you suggest then we can make discussions in the room. Also, having discussions here allows other people to review our work too. 
@bizzbyster

**Comment by bizzbyster** [comment]
sounds good

**Comment by KashishJagyasi** [comment]
creating discussions in the Room itself?
@bizzbyster

**Comment by ear-dev** [comment]
@KashishJagyasi can you show what it looks like when you click the plus sign in the last case you posted....... when the camera and the discussions are also behind it?  Thanks.

**Comment by KashishJagyasi** [comment]
Here's the view of the chatroom with chat bubbles. However, it Won't recommend using chat bubbles here because as the survey suggests, most of our users use RC for professional purposes. The information circulated via RC rooms will be more of like long notices or announcements. The interface with the chat bubbles definitely looks cool and better than that without them but using them in such a case may cause more clutter. Chat bubbles can be avoided here to give more breathing space to the text and it will also avoid infinite scrolls in case of long messages. This argument is backed by the fact that other chatting platforms like slack and discord don't use chat bubbles. Telegram is not a good example to compare to at this point because our user base is entirely different from that of telegram. Telegram is more inclined towards being a friendly chatting platform that RC is not ( As I've already compared our user base in the competitor analysis issue #306 ). However, I can try making a more detailed list of pros and cons of chat bubbles vs no chat bubbles if you like but for now, I'd suggest we should go with the no chat bubbles interface. 
![Frame 56](https://user-images.githubusercontent.com/47213420/88865345-431fc580-d225-11ea-9475-925ea480e083.png)
@ear-dev 
@bizzbyster

**Comment by KashishJagyasi** [comment]
@bizzbyster 
I think I misunderstood your previous comment on the room discussions. 
Eric told me that you might be questioning the need of the discussions in the rooms. Well as far as I understand it is a good feature because as Flora's survey results suggest, that people stop using RC because they face difficulty finding the topics that concern them. Threads are a great way to keep the discussions related to one topic in the same place but they don't provide options like replying to a particular message, react on it, or create another thread inside that thread. For example, We may consider a channel dedicated to RC GSoC discussions. GSoC has a lot of ongoing projects, and we can create separate discussions under the GSoC channel covering all the projects that are going on. It helps in keeping similar information in one place. So basically IMO, a discussion is like a subchannel under a parent channel. Much similar to the relation of the project boards and issues in Github

**Comment by KashishJagyasi** [comment]
The three-dot menu consists of the following items
Game center
Discussions 
Threads
Members
Files
Room info 
Notification preferences
Mentions
Starred
Pinned
Auto translate
Keyboard shortcuts
Share
I've already suggested placements for threads, discussions, and mentions. 
This is a possible placement for the remaining ones. Please lemme know what you think of these. Also, don't mind the spacing and other things yet it's kinda just a rough lo-fi wireframe.

![Frame 58 (3)](https://user-images.githubusercontent.com/47213420/88932132-4f466a00-d29b-11ea-8c05-b4cf27c85f53.png)

Well IMO Auto-Translate will be better off in universal settings and keyboard shortcuts have no meaning in a mobile view so I think we'll just discard that.
@ear-dev

**Comment by KashishJagyasi** [comment]
I tested this design on 3 of my friends and figured out that they expected that typing on the search will find them threads and discussions including the typed keywords too. 
So maybe we should show these too. Is that possible with a PWA. This level of search?

![Frame 65](https://user-images.githubusercontent.com/47213420/89931715-368a6c80-dc2a-11ea-8ed4-9984fa9dddb6.png)
